### DISCUSSION TITLE
Failed to get Transaction Write Lock

### DISCUSSION THREAD

**Discussion by tehkyle** [ravendb/ravendb#18113] [Q&A]
What would be some reasons the database storage engine would be throwing the following timeout exception?

System.TimeoutException: Waited for 00:00:30 for transaction write lock, but could not get it, the tx is currently owned by thread # - .NET ThreadPool Worker, OS thread id: ####

We are seeing it on an environment under heavy load fairly often. Looking at the source code the 30s timeout isn't modifyable. There is not alot of history with this exception but hints that it might be related to disk write speed. Here is the crystal mark of the CB drive on the machine we see the exception on:
![image](https://github.com/ravendb/ravendb/assets/1372402/8b8ff5e4-6186-45ba-9200-ced4f5c27447)

Any help in a direction for further troubleshooting is appreciated.

**Comment by garayx** [comment]
hi

looking on the crystal mark you provided I would say that your disk is defective or used,
for example here is my sata ssd crystal mark, see the random benchmarks (the speed are way greater then you posted):
![image](https://github.com/ravendb/ravendb/assets/16625387/4b574080-70d3-4394-bd43-42b0fb957f82)

**Reply by tehkyle** [reply]
Is there a minimum Disk IO spec or benchmark recommendations for RavenDB?

**Reply by garayx** [reply]
that's directly depends on the server load, usually the recommendation are sata ssd or nvme ssd.

You can use the Cluster Dashboard IO Stats widget to determine if your disk is queuing operations (Disk Queue), uses maximum 
 disk throughput.
 
![image](https://github.com/ravendb/ravendb/assets/16625387/34420fc1-8c17-4535-9e47-1f706b733245)

Also usually when you are hitting disk limits, RavenDB will popup a "extremely slow write to disk" notification in studio, which means the disk is throttled
![image](https://github.com/ravendb/ravendb/assets/16625387/4c13973c-0953-413b-b049-91f00e91216e)
